src/settingsmanager.cpp

```diff
@@ -147,8 +148,18 @@ void SettingsManager::initSettings()
     m_kiwixServerIpAddress = m_settings.value("localKiwixServer/ipAddress", QString("0.0.0.0")).toString();
     m_monitorDir = m_settings.value("monitor/dir", QString("")).toString();
     m_moveToTrash = m_settings.value("moveToTrash", true).toBool();
-    QVariant defaultLang = QVariant::fromValue(QLocale::languageToString(QLocale().language()) + '|' + QLocale().name().split("_").at(0));
-    m_langList = m_settings.value("language", {defaultLang}).toList();
+    QString defaultLang = QLocale::languageToString(QLocale().language()) + '|' + QLocale().name().split("_").at(0);
+
+    QList<QString> defaultLangList; // Qt5 QList doesn't support supplying a constructor list, so use append() for Qt5+Qt6 compat
+    defaultLangList.append(defaultLang);
+
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    QVariant defaultLangVariant(defaultLangList); // Qt5 requires explicit conversion from QList to QVariant
+    m_langList = m_settings.value("language", defaultLangVariant).toList();
+#else
+    m_langList = m_settings.value("language", defaultLangList).toList();
+#endif
```

Comment:
Yes, it can be done without conditional compilation. Since there is otherwise no context about why the objects are initialized in a specific way to maintain Qt5&Qt6 compatibility, I added a comment which hopefully is clear.